src/window.c

```diff
@@ -190,6 +191,58 @@ int S2D_Show(S2D_Window *window) {
             window->on_mouse(event);
           }
           break;
+        case SDL_FINGERMOTION:
+         {
+            if (window->on_touch) {
+           float fx,dx, fy, dy; 
+           fx = e.tfinger.x * window->width; // TODO: Viewport in scaling
+           fy = e.tfinger.y * window->height;
+           dx = e.tfinger.dx * window->width;
+           dy = e.tfinger.dy * window->height;
+           S2D_GetMouseOnViewport(window, (int)fx, (int)fy, &mx, &my);
```

Comment:
Touch events seem to deliver not positions, but indications of how far in a given dimension the event occurred, normalized between 0 and 1
e.g. 0.5 of x, 0.3 of y
which then need to be translated into coordinates in the window/viewport

The TODO comment "Viewport in scaling" seems to be not relevant - it seems a consequence of scaling the window to a bigger width is then that touch events can occur at a -ve position - since the original 0 axis is no longer flush with the left of the display, that is therefore OK and I had simply gotten confused.